### PR TITLE
Add missing upgrade path for Request::isMethodSafe() deprecation

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 3.2
| Fixed tickets| #20990

### CODE REVIEW

**Review by xabbuh — CHANGES_REQUESTED** [review]
We need to update the upgrade guide for 4.0 too.

**Comment by chalasr** [comment]
@xabbuh done

File: UPGRADE-4.0.md [hunk 1]
```diff
@@ -151,6 +151,11 @@ HttpFoundation
     - `isInvalid`/`isSuccessful`/`isRedirection`/`isClientError`/`isServerError`
     - `isOk`/`isForbidden`/`isNotFound`/`isRedirect`/`isEmpty`
 
+ * The ability to check only for cacheable HTTP methods using `Request::isMethodSafe()` is
+   not supported anymore and throws an exception. Disable checking only for cacheable methods
```
Comment (xabbuh) [inline UPGRADE-4.0.md]: It won't throw an exception. It will simply return `false`.

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: UPGRADE-4.0.md (hunk 1 above)
Comment (chalasr) [inline UPGRADE-4.0.md]: It is what the deprecation warning says: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Request.php#L1506
Should we change it?

Comment (xabbuh) [inline UPGRADE-4.0.md]: I think that comment refers to passing `true` to the method.

Comment (chalasr) [inline UPGRADE-4.0.md]: I thought the argument would be uncommented as is so the exception would be thrown when omitting to pass the argument to `false` explicitly.

Comment (chalasr) [inline UPGRADE-4.0.md]: updated

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**